x/ccv/provider/keeper/proposal.go

```diff
@@ -193,49 +144,59 @@ func (k Keeper) StopConsumerChain(ctx sdk.Context, chainID string, closeChan boo
 				err := k.chanCloseInit(ctx, channelID)
 				if err != nil {
 					k.Logger(ctx).Error("channel to consumer chain could not be closed",
-						"chainID", chainID,
+						"consumerId", consumerId,
 						"channelID", channelID,
 						"error", err.Error(),
 					)
 				}
 			}
 		}
-		k.DeleteChainToChannel(ctx, chainID)
-		k.DeleteChannelToChain(ctx, channelID)
+		k.DeleteConsumerIdToChannelId(ctx, consumerId)
+		k.DeleteChannelIdToConsumerId(ctx, channelID)
 	}
 
 	// delete consumer commission rate
-	provAddrs := k.GetAllCommissionRateValidators(ctx, chainID)
+	provAddrs := k.GetAllCommissionRateValidators(ctx, consumerId)
 	for _, addr := range provAddrs {
-		k.DeleteConsumerCommissionRate(ctx, chainID, addr)
+		k.DeleteConsumerCommissionRate(ctx, consumerId, addr)
 	}
 
-	k.DeleteInitChainHeight(ctx, chainID)
-	k.DeleteSlashAcks(ctx, chainID)
-	k.DeletePendingVSCPackets(ctx, chainID)
+	k.DeleteInitChainHeight(ctx, consumerId)
+	k.DeleteSlashAcks(ctx, consumerId)
+	k.DeletePendingVSCPackets(ctx, consumerId)
 
-	k.DeleteTopN(ctx, chainID)
-	k.DeleteValidatorsPowerCap(ctx, chainID)
-	k.DeleteValidatorSetCap(ctx, chainID)
-	k.DeleteAllowlist(ctx, chainID)
-	k.DeleteDenylist(ctx, chainID)
-	k.DeleteMinStake(ctx, chainID)
-	k.DisableInactiveValidators(ctx, chainID)
+	k.DeleteTopN(ctx, consumerId)
+	k.DeleteValidatorsPowerCap(ctx, consumerId)
+	k.DeleteValidatorSetCap(ctx, consumerId)
+	k.DeleteAllowlist(ctx, consumerId)
+	k.DeleteDenylist(ctx, consumerId)
+	k.DeleteMinStake(ctx, consumerId)
+	k.DisableInactiveValidators(ctx, consumerId)
 
-	k.DeleteAllOptedIn(ctx, chainID)
-	k.DeleteConsumerValSet(ctx, chainID)
+	k.DeleteAllOptedIn(ctx, consumerId)
+	k.DeleteConsumerValSet(ctx, consumerId)
 
-	k.Logger(ctx).Info("consumer chain removed from provider", "chainID", chainID)
+	// TODO (PERMISSIONLESS) add newly-added state to be deleted
+
+	k.Logger(ctx).Info("consumer chain removed from provider", "consumerId", consumerId)
 
 	return nil
 }
 
-// MakeConsumerGenesis constructs the consumer CCV module part of the genesis state.
+// MakeConsumerGenesis returns the created consumer genesis state for consumer chain `consumerId`,
+// as well as the validator hash of the initial validator set of the consumer chain
 func (k Keeper) MakeConsumerGenesis(
 	ctx sdk.Context,
-	prop *types.ConsumerAdditionProposal,
+	consumerId string,
 ) (gen ccv.ConsumerGenesisState, nextValidatorsHash []byte, err error) {
-	chainID := prop.ChainId
+	initializationRecord, found := k.GetConsumerIdToInitializationRecord(ctx, consumerId)
+	if !found {
+		return gen, nil, errorsmod.Wrapf(types.ErrNoInitializationRecord,
+			"initialization record for consumer chain: %s is missing", consumerId)
+
+	}
+	updateRecord := k.GetConsumerIdToUpdateRecordOrDefault(ctx, consumerId, DefaultUpdateRecord)
```

Comment:
If there is no updateRecord, then just set TopN to default as it's the only field used here.